packages/recommendations-vdom/src/types/Renderer.ts

```diff
@@ -0,0 +1,36 @@
+export type Pragma = (
+  type: any,
+  props: Record<string, any> | null,
+  ...children: ComponentChildren[]
+) => JSX.Element;
```

Comment:
where does JSX get defined?

Reply:
It's a TypeScript global, we also use it in Autocomplete.